plugin/evm/customheader/time_test.go

```diff
@@ -15,81 +15,139 @@ import (
+			name: "granite_time_milliseconds_earlier_than_parent_should_fail",
+			header: generateHeader(
+				timeSeconds,
+				utils.NewUint64(timeSeconds*1000),
+			),
+			parentHeader: generateHeader(
+				timeSeconds,
+				utils.NewUint64(timeSeconds*1000+1),
+			),
+			extraConfig: extras.TestGraniteChainConfig,
+			expectedErr: errBlockTooOld,
 		},
 		{
-			name:             "granite_time_milliseconds_matching_time_should_work",
-			timeSeconds:      timeSeconds,
-			timeMilliseconds: utils.NewUint64(timeSeconds * 1000),
-			extraConfig:      extras.TestGraniteChainConfig,
-			expectedErr:      nil,
+			name:        "pre_granite_time_too_far_in_future_should_fail",
+			header:      generateHeader(uint64(now.Add(MaxFutureBlockTime).Add(1*time.Second).Unix()), nil),
+			extraConfig: extras.TestFortunaChainConfig,
+			expectedErr: ErrBlockTooFarInFuture,
 		},
 		{
-			name:             "granite_time_milliseconds_matching_time_rounded_should_work",
-			timeSeconds:      timeSeconds,
-			timeMilliseconds: utils.NewUint64(timeMillis),
-			extraConfig:      extras.TestGraniteChainConfig,
-			expectedErr:      nil,
+			name: "granite_time_too_far_in_future_should_fail",
+			header: generateHeader(
+				uint64(now.Add(MaxFutureBlockTime).Add(1*time.Second).Unix()),
+				utils.NewUint64(uint64(now.Add(MaxFutureBlockTime).Add(1*time.Second).UnixMilli())),
+			),
+			extraConfig: extras.TestGraniteChainConfig,
+			expectedErr: ErrBlockTooFarInFuture,
 		},
 		{
-			name:             "granite_time_milliseconds_less_than_time_should_fail",
-			timeSeconds:      timeSeconds,
-			timeMilliseconds: utils.NewUint64(timeSeconds*1000 - 1),
-			extraConfig:      extras.TestGraniteChainConfig,
-			expectedErr:      ErrTimeMillisecondsMismatched,
+			name: "granite_time_milliseconds_too_far_in_future_should_fail",
+			header: generateHeader(
+				uint64(now.Add(MaxFutureBlockTime).Unix()),
+				utils.NewUint64(uint64(now.Add(MaxFutureBlockTime).Add(1*time.Millisecond).UnixMilli())),
+			),
+			extraConfig: extras.TestGraniteChainConfig,
+			expectedErr: ErrBlockTooFarInFuture,
 		},
 		{
-			name:             "granite_time_milliseconds_greater_than_time_should_fail",
-			timeSeconds:      timeSeconds,
-			timeMilliseconds: utils.NewUint64(timeSeconds * 1001),
-			extraConfig:      extras.TestGraniteChainConfig,
-			expectedErr:      ErrTimeMillisecondsMismatched,
+			name:         "first_granite_block_should_work",
+			header:       generateHeader(timeSeconds, utils.NewUint64(timeMillis)),
+			parentHeader: generateHeader(timeSeconds, nil),
+			extraConfig:  extras.TestGraniteChainConfig,
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			header := customtypes.WithHeaderExtra(
-				&types.Header{
-					Time: test.timeSeconds,
-				},
-				&customtypes.HeaderExtra{
-					TimeMilliseconds: test.timeMilliseconds,
-				},
-			)
-			err := VerifyTime(test.extraConfig, header, time)
+			// if parentHeader is nil, copy the current header
```

Comment:
If I was Arran reviewing, I'd say this comment just describes what the code does and isn't useful - what's the "why," rather than the "what"?